File: app/src/main/java/org/mesonet/app/maps/MapsModel.java
package org.mesonet.app.maps;


import com.google.gson.annotations.SerializedName;

import java.util.List;
import java.util.Map;

public class MapsModel
{
    @SerializedName("main")
    private List<MainModel> mMain;

    @SerializedName("sections")
    private Map<String, SectionModel> mSections;

    @SerializedName("products")
    private Map<String, ProductModel> mProducts;


    public List<MainModel> GetMain()
    {
        return mMain;
    }



    public Map<String, SectionModel> GetSections()
    {
        return mSections;
    }



    public Map<String, ProductModel> GetProducts()
    {
        return mProducts;
    }


    public static class MainModel
    {
        @SerializedName("sections")
        private List<String> mSections;

        @SerializedName("title")
        private String mTitle;



        public List<String> GetSections()
        {
            return mSections;
        }



        public String GetTitle()
        {
            return mTitle;
        }
    }



    public static class SectionModel
    {
        @SerializedName("title")
        public String mTitle;

        @SerializedName("products")
        private List<String> mProducts;



        public String GetTitle()
        {
            return mTitle;
        }



        public List<String> GetProducts()
        {
            return mProducts;
        }
    }



    public static class ProductModel
    {
        @SerializedName("url")
        public String mUrl;

        @SerializedName("title")
        public String mTitle;
    }
}

File: app/src/main/java/org/mesonet/app/maps/MapsModel.kt
package org.mesonet.app.maps


import com.google.gson.annotations.SerializedName

class MapsModel {
    @SerializedName("main")
    private val mMain: List<MainModel>? = null

    @SerializedName("sections")
    private val mSections: Map<String, SectionModel>? = null

    @SerializedName("products")
    private val mProducts: Map<String, ProductModel>? = null


    fun GetMain(): List<MainModel>? {
        return mMain
    }


    fun GetSections(): Map<String, SectionModel>? {
        return mSections
    }


    fun GetProducts(): Map<String, ProductModel>? {
        return mProducts
    }


    class MainModel {
        @SerializedName("sections")
        private val mSections: List<String>? = null

        @SerializedName("title")
        private val mTitle: String? = null


        fun GetSections(): List<String>? {
            return mSections
        }


        fun GetTitle(): String? {
            return mTitle
        }
    }


    class SectionModel {
        @SerializedName("title")
        var mTitle: String? = null

        @SerializedName("products")
        private val mProducts: List<String>? = null


        fun GetTitle(): String? {
            return mTitle
        }


        fun GetProducts(): List<String>? {
            return mProducts
        }
    }


    class ProductModel {
        @SerializedName("url")
        var mUrl: String? = null

        @SerializedName("title")
        var mTitle: String? = null
    }
}
